keanu-project/src/test/java/io/improbable/keanu/vertices/dbl/nonprobabilistic/operators/TensorTestOperations.java

```diff
@@ -0,0 +1,72 @@
+package io.improbable.keanu.vertices.dbl.nonprobabilistic.operators;
+
+import static org.junit.Assert.assertEquals;
+
+import java.util.List;
+
+import io.improbable.keanu.tensor.Tensor;
+import io.improbable.keanu.tensor.dbl.DoubleTensor;
+import io.improbable.keanu.vertices.dbl.DoubleVertex;
+
+public class TensorTestOperations {
+
+    public static void finiteDifferenceMatchesGradient(List<DoubleVertex> inputVertices,
+                                                       DoubleVertex outputVertex,
+                                                       final double INCREMENT_AMOUNT,
+                                                       final double DELTA) {
```

Comment:
Why the use of `final` here?

We're pretty lax with our use of `final` for local variables, I'm sure there's thousands of places in the code base where we could use it but we're not.

Reply:
It's just a bit of a pet hate of mine tbh.  I find a wider use of final (const) can be annoying if the rest of your ecosystem doesn't deal with it well, but if the use is consistent it can lead to better quality code (and also helps the compiler/JIT produce better code too).  I'd prefer to leave these  unless you have serious objections?

Reply:
No major objection to the `final`. I find the ALL_CAPS a bit jarring, and I don't think this is the convention for final parameter variables? I think ALL_CAPS is only used for static final member variables. But this is really splitting hairs.